rest/admin_api.go

```diff
@@ -1403,14 +1409,16 @@ func (h *handler) getUsers() error {
 }
 
 func (h *handler) getRoles() error {
+	includeDeleted, _ := h.getOptBoolQuery(paramDeleted, false)
+
 	var roles []string
 	var err error
 
-	includeDeleted, _ := h.getOptBoolQuery(paramDeleted, false)
-	if includeDeleted {
+	// GetRoleIDs supports Views only when includeDeleted=false
```

Comment:
As discussed, move the conditional logic down to the db package so that it can have ownership over which view/index to choose to use for a given request.